### PR TITLE
Commerce 2908 + Commerce 2912 Account selector and global search refactor

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
From #291 and #381 /cc @gianmarcobrunialti @ivicac

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

**Comment by alecsloan** [comment]
@wincent I made most of the changes you requested in #381. However there were 3 things I wasn't able to do and @gianmarcobrunialti @FabioDiegoMastrorilli or @garofalog will have to change:

- Items.map((item) => {}) to items.map(() => {}) in AccountSelector.js
- Replace useRef with useEventListener in GlobalSearch.js
- Remove Faker

@FabioDiegoMastrorilli you can close #291 and #381

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

**Comment by alecsloan** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/model/OrderStatusModel.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * @author Luca Pellizzon
+ * @author Fabio Diego Mastrorilli
+ */
+public class OrderStatusModel {
+
+	public OrderStatusModel(int code, String label, String localizedLabel) {
+		_code = code;
+		_label = label;
+
+		_label_i18n = localizedLabel;
```
Comment (dsitu) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/model/OrderStatusModel.java:27]:
```suggestion
		_label = label;
		_label_i18n = localizedLabel;
```

**Review by dsitu — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/model/OrderStatusInfo.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @author Luca Pellizzon
+ * @author Fabio Diego Mastrorilli
+ */
+public class OrderStatusInfo {
+
+	public OrderStatusInfo(int code, String label, String localizedLabel) {
+		_code = code;
+		_label = label;
+
+		_label_i18n = localizedLabel;
```
[the remark this answers is not in the document]
Comment (alecsloan) [reply modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/model/OrderStatusInfo.java:27]: It looks like auto sf adds the space

[the remark this answers is not in the document]
Comment (dsitu) [reply modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/cart/model/OrderStatusInfo.java:27]: ahh, ok, all good then.. ahha

**Review by dsitu — APPROVED** [review]

**Review by wincent — COMMENTED** [review]
Changes in d7295429902542860e13d6c9dffc8b797525ea43 look good to me.

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/model/CurrentAccountModel.java [hunk 3]
```diff
@@ -19,26 +19,26 @@
  */
 public class CurrentAccountModel {
 
-	public CurrentAccountModel(long accountId, String name, String thumbnail) {
-		_accountId = accountId;
+	public CurrentAccountModel(long id, String name, String logoURL) {
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/model/CurrentAccountModel.java:22]: SF order

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AccountSelectorTag.java [hunk 4]
```diff
@@ -93,79 +119,75 @@ public int doStartTag() {
...
-			putValue(
-				"createNewOrderLink", _getAddCommerceOrderURL(themeDisplay));
+				httpServletRequest.setAttribute(
+					"liferay-commerce:account-selector:currentOrder",
+					currentOrderModel);
+			}
 
-			putValue(
-				"viewAllOrdersLink", _getViewCommerceOrdersURL(themeDisplay));
+			String setCurrentAccountURL =
```
Comment (ivicac) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AccountSelectorTag.java:146]: Move closer where it is used

**Review by ivicac — CHANGES_REQUESTED** [review]

**Comment by ivicac** [comment]
@gianmarcobrunialti please check @alecsloan's comments regarding the javascript part plus there are also issues with tests(there is an interdependency between them)

**Comment by alecsloan** [comment]
ci:test:sf

**Review by marco-leo — APPROVED** [review]

**Comment by alecsloan** [comment]
ci:test:sf

**Comment by alecsloan** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:forward

**Pull request closed without merging**